File: 649.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        n = len(senate)
        r_queue, d_queue = [], []
        for i, s in enumerate(senate):
            if s == 'R':
                r_queue.append(i)
            else:
                d_queue.append(i)
        while r_queue and d_queue:
            if r_queue[0] < d_queue[0]:
                r_queue.append(r_queue[0] + n)
            else:
                d_queue.append(d_queue[0] + n)
            r_queue.pop(0)
            d_queue.pop(0)
        return 'Radiant' if r_queue else 'Dire'

File: cpp/649.cpp
#include<iostream>
#include<queue>
#include<string>

using namespace std;

class Solution {
public:
    string predictPartyVictory(string senate) {
        int n = senate.size();
        queue<int> rQueue, dQueue;
        for(int i = 0; i < n; i++){
            if(senate[i] == 'R') rQueue.push(i);
            else dQueue.push(i);
        }
        while(!rQueue.empty() && !dQueue.empty()){
            if(rQueue.front() < dQueue.front())
                rQueue.push(rQueue.front() + n);
            else dQueue.push(dQueue.front() + n);
            rQueue.pop();
            dQueue.pop();
        }
        return !rQueue.empty() ? "Radiant" : "Dire";
    }
};